rust/crd/src/lib.rs

```diff
@@ -109,32 +112,45 @@ pub enum Error {
     )
 )]
 pub struct HiveClusterSpec {
-    /// General Hive metastore cluster settings
+    // no doc - docs in HiveClusterConfig struct.
     pub cluster_config: HiveClusterConfig,
-    /// Cluster operations like pause reconciliation or cluster stop.
+
+    // no doc - docs in ClusterOperation struct.
     #[serde(default)]
     pub cluster_operation: ClusterOperation,
-    /// The Hive metastore image to use
+
+    // no doc - docs in ProductImage struct.
     pub image: ProductImage,
+
+    // no doc - docs in Role struct.
     #[serde(default, skip_serializing_if = "Option::is_none")]
     pub metastore: Option<Role<MetaStoreConfigFragment>>,
 }
 
+/// Hive metastore settings that affect all roles and roll groups.
+/// This includes the metastore database connection details, the HDFS and S3 connection,
+/// log aggregation configuration and service exposition settings.
```

Comment:
Can we go for a more generic description here? The cluster config maybe changing which means adapting the comment. I would rather focus on why there is a cluster config (e.g. because all roles / rolegroups will connect to the same database etc for HA and therefore this does not have to be configurable on role / group level).

Reply:
https://github.com/stackabletech/hive-operator/pull/394/commits/ac94c3848545c918feea62434e4025ddab75b74b